src/AutoRest.CSharp.V3/CodeGen/BaseCSharpWriter.cs

```diff
@@ -6,30 +6,27 @@
 
 namespace AutoRest.CSharp.V3.CodeGen
 {
-    public abstract class BaseCSharpWriter
+    internal abstract class BaseCSharpWriter
     {
-        private HashSet<string> _usings = new HashSet<string>();
+        private readonly HashSet<string> _usings = new HashSet<string>();
         public BaseCSharpWriter()
         {
         }
 
         public class CodeBlock : IDisposable
         {
-            public Action? _endBlockAction;
+            public Action? EndBlockAction;
```

Comment:
Sorry this should have been a private field not public. In which case using the "_" made sense.

Reply:
I'll be cleaning this stuff up later anyway as there are several things that need to be worked on/fixed. I'm not worried about it at the moment.